gemini/prompts/prompt_optimizer/vertex_ai_prompt_optimizer_sdk_custom_metric.ipynb

```diff
@@ -0,0 +1,1390 @@
+        "Create a storage bucket to store intermediate artifacts such as datasets."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "MzGDU7TWdts_"
+      },
+      "outputs": [],
+      "source": [
+        "BUCKET_NAME = \"your-bucket-name-{PROJECT_ID}-unique\"  # @param {type:\"string\"}\n",
+        "\n",
+        "BUCKET_URI = f\"gs://{BUCKET_NAME}\"  # @param {type:\"string\"}"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "NIq7R4HZCfIc"
+      },
+      "outputs": [],
+      "source": [
+        "! gsutil mb -l {REGION} -p {PROJECT_ID} {BUCKET_URI}"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "set_service_account"
+      },
+      "source": [
+        "#### Service Account and permissions\n",
+        "\n",
+        "Vertex AI Automated Prompt Design requires a service account with the following permissions:\n",
+        "\n",
+        "-   `Vertex AI User` to call Vertex LLM API\n",
+        "-   `Storage Object Admin` to read and write to your GCS bucket.\n",
+        "-   `Artifact Registry Reader` to download the pipeline template from Artifact Registry.\n",
+        "-   `Cloud Run Developer` to deploy function on Cloud Run.\n",
+        "\n",
+        "[Check out the documentation](https://cloud.google.com/iam/docs/manage-access-service-accounts#iam-view-access-sa-gcloud) to know how to grant those permissions to a single service account.\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "ssUJJqXJJHgC"
+      },
+      "outputs": [],
+      "source": [
+        "SERVICE_ACCOUNT = f\"{PROJECT_NUMBER}-compute@developer.gserviceaccount.com\""
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "wqOHg5aid6HP"
+      },
+      "outputs": [],
+      "source": [
+        "! gcloud projects add-iam-policy-binding {PROJECT_ID} \\\n",
+        "    --member=serviceAccount:{SERVICE_ACCOUNT} \\\n",
+        "    --role=roles/aiplatform.user\n",
+        "\n",
+        "! gcloud projects add-iam-policy-binding {PROJECT_ID} \\\n",
+        "    --member=serviceAccount:{SERVICE_ACCOUNT} \\\n",
+        "    --role=roles/storage.objectAdmin\n",
+        "\n",
+        "! gcloud projects add-iam-policy-binding {PROJECT_ID} \\\n",
+        "    --member=serviceAccount:{SERVICE_ACCOUNT} \\\n",
+        "    --role=roles/artifactregistry.reader\n",
+        "\n",
+        "! gcloud projects add-iam-policy-binding {PROJECT_ID} \\\n",
+        "    --member=serviceAccount:{SERVICE_ACCOUNT} \\\n",
+        "    --role=roles/run.developer\n",
+        "\n",
+        "! gcloud projects add-iam-policy-binding {PROJECT_ID} \\\n",
+        "    --member=serviceAccount:{SERVICE_ACCOUNT} \\\n",
+        "    --role=roles/run.invoker"
```

Comment:
Is there a way to combine these into one command to make it less verbose?

Reply:
done